Toolchain/Patches/llvm.patch

```diff
@@ -872,11 +831,33 @@ index 171318ff6..aa7345a19 100644
      #define _LIBUNWIND_USE_DLADDR 1
    #else
      #define _LIBUNWIND_USE_DLADDR 0
+diff --git a/llvm/cmake/modules/HandleLLVMOptions.cmake b/llvm/cmake/modules/HandleLLVMOptions.cmake
+index 0c3419390..0cfea47a4 100644
+--- a/llvm/cmake/modules/HandleLLVMOptions.cmake
++++ b/llvm/cmake/modules/HandleLLVMOptions.cmake
+@@ -228,11 +228,12 @@ endif()
+ 
+ # Pass -Wl,-z,defs. This makes sure all symbols are defined. Otherwise a DSO
+ # build might work on ELF but fail on MachO/COFF.
+-if(NOT (CMAKE_SYSTEM_NAME MATCHES "Darwin|FreeBSD|OpenBSD|DragonFly|AIX|SunOS|OS390" OR
+-        WIN32 OR CYGWIN) AND
+-   NOT LLVM_USE_SANITIZER)
+-  set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-z,defs")
+-endif()
++# TODO: get our own CMAKE_SYSTEM_NAME option somehow
```

Comment:
CMAKE_SYSTEM_NAME comes from the toolchain file, and works b/c the toolchain has Toolchain/CMake/Platform/SerenityOS.cmake in the CMAKE_MODULE_PATH. (again with the future looking comments :)). Ideally we could put SerenityOS in the "no ty" list here.

Reply:
We *should* be able to pass `CMAKE_SYSTEM_NAME` to the LLVM build so that should work. Note that we can't use the cross-compiler toolchain file as that compiler does not exist when we are building it :P

Reply:
Also
```
[llvm/build] System is unknown to cmake, create:
[llvm/build] Platform/SerenityOS to use this system, please post your config file on discourse.cmake.org so it can be added to cmake
```